Source/Libraries/FaultData/DataResources/FaultDataResource.cs

```diff
@@ -1023,14 +1023,80 @@ private void PopulateFaultInfo(Fault fault, DataGroup dataGroup, VICycleDataGrou
 
             if (calculationCycle >= 0)
             {
+                TableOperations<Meter> meterTable = new TableOperations<Meter>(connection);
+                string testQuery = "ID={0} AND LocationID in (Select LocationID from AssetLocation where AssetID={1})";
                 for (int i = 0; i < fault.Curves.Count; i++)
                 {
                     Fault.Summary summary = new Fault.Summary();
                     summary.DistanceAlgorithmIndex = i;
                     summary.DistanceAlgorithm = fault.Curves[i].Algorithm;
                     summary.Distance = fault.Curves[i][calculationCycle].Value;
-                    summary.IsValid = IsValid(summary.Distance, dataGroup);
+                    Line line = Line.DetailedLine(dataGroup.Asset);
+                    List<Fault.Path> faultPaths = new List<Fault.Path>();
+                    for (int index = 0; index < line.Path.Count; index++)
+                    {
+                        // We can break early if this condition is found because any segments after are shorter
+                        if (summary.Distance > line.Path[index].Length) break;
+                        Fault.Path forwardPath = new Fault.Path();
+                        forwardPath.TransmissionPath = line.Path[index];
+                        forwardPath.PathNumber = index;
+                        forwardPath.FaultDistance = summary.Distance;
+                        forwardPath.IsMain = false;
+                        // Skip if path doesn't exist
+                        bool useForward = true;
+                        bool useBackward = false;
+                        if (forwardPath.TransmissionPath.Segments.Count == 0) continue;
+
+                        // If we have exactly one segment then a backwards path won't be neccessary
+                        if (forwardPath.TransmissionPath.Segments.Count != 1)
+                        {
+                            bool forwardlyChiral= meterTable.QueryRecordsWhere(testQuery, meter.ID, forwardPath.TransmissionPath.Segments[0].ID).Any();
+                            bool backwardlyChiral = meterTable.QueryRecordsWhere(testQuery, meter.ID, forwardPath.TransmissionPath.Segments[forwardPath.TransmissionPath.Segments.Count - 1].ID).Any();
+                            useForward = forwardlyChiral || !backwardlyChiral;
+                            useBackward = !forwardlyChiral || backwardlyChiral;
```

Comment:
Are we envisioning some kind of looped transmission line where the meter is somehow monitoring from both ends of the path simultaneously? It shouldn't be physically possible for a single set of currents to represent current flowing from two physically distinct line segments. Seems to me like `useForward && useBackward` should represent an error, and the path should only be traversed in one direction.

Reply:
What I am trying to accomplish is to log every possible line segment the fault could be on. Therefore, if there is uncertainty on which direction we need to go, we need to log both.